staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -253,7 +253,12 @@ func (flags *ApplyFlags) ToOptions(cmd *cobra.Command, baseName string, args []s
 		return nil, err
 	}
 
-	deleteOptions, err := flags.DeleteFlags.ToOptions(dynamicClient, flags.IOStreams)
+	discovery, err := flags.Factory.ToDiscoveryClient()
```

Comment:
It seems strange that `DiscoveryClient` is now required for `apply`, `replace`, and `run`.  I guess that is because they use `DeleteFlags`.

Reply:
Yes, exactly because of that.